apps/admin_api/lib/admin_api/v1/controllers/token_controller.ex

```diff
@@ -82,15 +82,31 @@ defmodule AdminAPI.V1.TokenController do
       |> Map.put("account_uuid", Account.get_master_account().uuid)
       |> Token.insert()
 
-    case attrs["amount"] do
-      amount when is_number(amount) and amount > 0 ->
-        inserted_token
-        |> MintGate.mint_token(%{"amount" => amount})
-        |> respond_single(conn)
+    create_with_mint(attrs["amount"], inserted_token, conn)
```

Comment:
Can we validate for a valid amount and return error, even before token is inserted?